pkg/kubectl/cmd/util/factory.go

```diff
@@ -169,7 +169,8 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 				return kubectl.MakeLabels(t.Labels), nil
 			case *api.Service:
 				if t.Spec.Selector == nil {
-					return "", fmt.Errorf("the service has no pod selector set")
+					// External services should be exposable; this is not an error
```

Comment:
Well, technically exposing a service -> a service only works if you copy the mechanism by which service 1 shows stuff.  So this is an error in Kube.  But in openshift, where we expose a route from a service, there is no need to for the service to have a selector (we're not creating a new service there).  I think this is just a change to OpenShift